common/beerocks/bwl/shared/nl_genl_socket.h

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+#ifndef __BWL_NL_GENL_SOCKET_H__
+#define __BWL_NL_GENL_SOCKET_H__
+
+#include "netlink_socket.h"
+
+namespace bwl {
+
+/**
+ * @brief C++ wrapper class around the generic Netlink protocol library (libnl-genl).
+ *
+ * This is a very simple C++ wrapper class around the generic Netlink protocol library. Its aim is
+ * to facilitate the use of the library from C++ applications. This class extends the
+ * netlink_socket base class for the generic Netlink protocol (NETLINK_GENERIC).
+ *
+ * Known derived classes: nl80211_socket.
+ */
```

Comment:
With no functionality other than setting the protocol, I'm not sure if it's worthwhile to introduce this additional class. IMO it makes the code harder to understand with no benefit at all.

This class is only useful if later on 
- there is a second, non-nl80211 netlink socket (and there probably should be one, for nlrt, but that one will not be in bwl so code needs to be refactored anyway);
- *and* there is some common code between them that should be in this class and not in netlink_socket itself.

But in that case, we can just introduce the class at that time.